server/src/main/java/org/opensearch/action/search/SearchRequestContext.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.lucene.search.TotalHits;
+
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class holds request-level context for search queries at the coordinator node
+ *
+ * @opensearch.internal
+ */
+public class SearchRequestContext {
+    private final SearchRequestOperationsListener searchRequestOperationsListener;
+    private long absoluteStartNanos;
+    private final Map<String, Long> phaseTookMap;
+    private TotalHits totalHits;
+    private final EnumMap<ShardStatsFieldNames, Integer> shardStats;
+
+    public SearchRequestContext() {
+        this(new SearchRequestOperationsListener.CompositeListener(List.of(), LogManager.getLogger()));
+    }
```

Comment:
This constructor is only used by tests. Can we please make it package-private (i.e. remove the `public` modifier) and add a Javadoc saying something like "For testing only".

Reply:
Added and made package private